src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -167,6 +167,29 @@ public SnapshotService(
     this.snapshotBuilderSettingsDao = snapshotBuilderSettingsDao;
   }
 
+  public String generateUpdatedSnapshotNameIfByRequest(SnapshotRequestModel model) {
+    SnapshotRequestContentsModel contentsModel = model.getContents().get(0);
+    if (contentsModel.getMode() == SnapshotRequestContentsModel.ModeEnum.BYREQUESTID) {
+      SnapshotAccessRequestResponse snapshotAccessRequestResponse =
+          snapshotRequestDao.getById(contentsModel.getRequestIdSpec().getSnapshotRequestId());
+      String dashesAndSpacesRegex = "[- ]+";
+      String nonAlphaNumericRegex = "\\W";
+
+      return org.springframework.util.StringUtils.trimTrailingCharacter(
+          org.springframework.util.StringUtils.trimLeadingCharacter(
+              String.format(
+                      "%s_%s",
+                      snapshotAccessRequestResponse.getSnapshotName(),
```

Comment:
do we need to be worried about entering user entered information like the name into the database?

Reply:
okay cool, just checking, is that enough to ensure safety? I just remember the security trainings saying regex matching may not be the most complete solution

Reply:
I think there's extra validation that gets done already, since this is just generating a name for `byRequestId` - or at least, I would hope. This doesn't bypass any existing name checks or anything, so it is no less safe than we have already.

Reply:
Sounds good, not a problem for this ticket. But if there is not validation somewhere, we should file a ticket.